Fixed inline validation for Archetypes content. [4.3]

mauritsvanrees opened the pull request:
Every field was shown as having an error, without showing an error message.
Fixes https://github.com/plone/Products.CMFPlone/issues/1982

The wrong behavior was a side effect of the base tag changes in Plone 4.3.12.

Notes:
- dexterity is not affected, because it uses z3c form inline validation
- I did not see errors when viewing PloneFormGen forms, it does not have the `blurrable` class that would be needed for inline validation to kick in.

rodfersou commented:
@mauritsvanrees your implementation works, but maybe it is better to use `location.href` instead of `<base>` tag.

We keep the `<base>` tag at Plone 4.3.x just to minimize the problems, but the ideal was to remove this tag the same way we do at Plone 5.x

mauritsvanrees commented:
True. But `location.href` can contain query parameters, so we would need to strip those off as well. That goes fine with my code as long as the url actually contains 'edit', which would usually be the case for `location.href` as well, but not always.

Actually, I think we can do one better. The formlib and z3c.form validations that are in this same file, use the url of the form action. I have updated the pull request to use this for archetypes as well.

And I have added an extra check: if the response we get has an undefined error message, as happens when you get html back instead of a json message, then we don't display an error.

rodfersou commented:
Just for the sake of curiosity, this gist show how to parse the URL, then you can remove the query parameters easily https://gist.github.com/jlong/2428561

mauritsvanrees commented:
Ah, you pointed to something similar in the issue, but I didn't catch on to how I could use that. I'll try to remember this one for the next time I need something like this. Thanks!
